todo-list/script.js

```diff
@@ -1,150 +1,314 @@
-// => State
-let todos = [];
-
-// Select Elements
-const app = document.querySelector('.todo-container');
-const inputTodo = app.querySelector('#add-todo');
-const addBtnTodo = app.querySelector('.wrapper-add .add-btn');
-const wrapperList = app.querySelector('.wrapper-list');
-const description = app.querySelector('.description');
-const clearButton = app.querySelector('.wrapper-description .clear-all');
-
-
-// => functions
-function renderTodos() {
-    const completedTask = todos.filter(todoObj => todoObj.checked == true).length;
-    description.innerText = `Completed tasks ${completedTask}`;
-    let todoString = '';
-    wrapperList.innerHTML = '';
-
-    completedTask > 0 ? clearButton.hidden = false : clearButton.hidden = true;
-
-    todos.forEach((todoObj, index) => {
-        todoString += `
-            <li id='${index}'>
-                <label class="todo-wrapper" for="task${index}">
-                    <div class="flex-left">
-                        <input type="checkbox" name="todo-task" id="task${index}" class="todo-checkbox" ${todoObj.checked ? 'checked' : ''}>
-                        <span class="todo-text">${todoObj.label}</span>
-                        <span class="checkmark"></span>
-                        <input type="text" class="todo-edit" value="${todoObj.label}" hidden>
-                    </div>
-                    <div class="flex-right">
-                        <button class="delete-todo">&#8999;</button>
-                        <button class="edit-todo">&#10247;</button>
-                    </div>
-                </label>
-            </li>
-        `;
-        wrapperList.innerHTML = todoString;
-    });
-    
-    saveToLocalStorageState();
+class Task {
+    constructor(label, id) {
+        this.label = label;
+        this.isCompeleted = false;
+        this.taskId = id;
+    }
 }
 
-function addTodo(event) {
-    event.preventDefault();
-    const label = this.value.trim();
-    let checked = false;
-    if(this.value != '' && !todos.find(todo => todo == this.value)) {
-        todos = [
-            ...todos,
-            {
-                label: this.value,
-                checked
+class TodoList {
+    
+    constructor(elementInitApp, preset = {}) {
+        this.titleOfApp = preset.title || 'Todo List App';
```

Comment:
You may consider using [Object.assing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) to assign default params